test/OpenTelemetry.Tests/Implementation/Trace/Propagation/CustomPropagator1.cs

```diff
@@ -0,0 +1,45 @@
+﻿// <copyright file="CustomPropagator1.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using OpenTelemetry.Context.Propagation;
+
+namespace OpenTelemetry.Tests.Implementation.Trace.Propagation
+{
+    public class CustomPropagator1 : ITextFormat
```

Comment:
`CustomPropagator1` & `CustomPropagator2` are a bit different than what I was expecting. Why not just have a single `TestPropagator` that you can pass in the header name that it will use for propagation? Should allow you to model everything you need for test, and fully implement the API (like Fields).

Reply:
@eddynaka Can you make this suggestion?

Reply:
Hi @cijothomas , i'm talking with @CodeBlanch , to understand the idea.

Reply:
@cijothomas , can you check again? @CodeBlanch helped me with the refactor.